### ISSUE TITLE
CustomDatasetFromCSV.__getitem__ Lacks Vectorization

### ISSUE THREAD

**Issue by jacobeturpin** [utkuozbulak/pytorch-custom-dataset-examples#2]
The current approach of reshaping an image in the `CustomDatasetFromCSV` class uses a slow for loop rather than a vectorized implementation.

See the following reshape logic in `CustomDatasetFromCSV.__getitem__`:
```python
img_as_np = np.ones((28, 28), dtype='uint8')
# Fill the numpy array with data from pandas df
for i in range(1, self.data.shape[1]):
    row_pos = (i-1) // self.height
    col_pos = (i-1) % self.width
    img_as_np[row_pos][col_pos] = self.data.iloc[index][i]
```

Best practice here should probably be to use `np.reshape`. 

When doing something very simple like loading a CSV copy of the MNIST dataset, this can cause a significant slowdown (e.g. my local machine training a very simple CNN on pytorch-cpu can take <1 min for an epoch with a vectorized approach versus >30 min with the current example).

**Comment by utkuozbulak** [comment]
Hello, thanks for the feedback! I will have a look.

**Comment by utkuozbulak** [comment]
Addressed by #3 thanks to @PyeongKim
